console/console-init/ui/src/modules/device/components/DeviceFilter/DeviceFilter.tsx

```diff
@@ -0,0 +1,388 @@
+    const filterObj = { ...filter };
+    if (value != filterObj.deviceId) {
+      filterObj.deviceId = value;
+      setFilter(filterObj);
+    }
+  };
+
+  const onTypeSelect = (value: string) => {
+    const filterObj = { ...filter };
+    filterObj.deviceType = value;
+    setFilter(filterObj);
+  };
+
+  const onStatusSelect = (value: string) => {
+    const filterObj = { ...filter };
+    filterObj.status = value;
+    setFilter(filterObj);
+  };
+
+  const addCriteria = () => {
+    const filterObj = { ...filter };
+    const list = filterObj.filterCriteria;
+    list.push({
+      operator: operator.eq,
+      parameter: "",
+      value: "",
+      key: uniqueId()
+    });
+    filterObj.filterCriteria = list;
+    setFilter(filterObj);
+  };
+
+  const deleteCriteria = (criteria: IDeviceFilterCriteria) => {
+    const filterObj = { ...filter };
+    const list = filterObj.filterCriteria;
+    const index = findIndexByProperty(list, "key", criteria.key);
+    list.splice(index, 1);
+    filterObj.filterCriteria = list;
+    setFilter(filterObj);
+  };
+
+  const updateCriteria = (criteria: IDeviceFilterCriteria) => {
+    const filterObj = { ...filter };
+    const list = filterObj.filterCriteria;
+    const index = findIndexByProperty(list, "key", criteria.key);
+    list[index] = criteria;
+    filterObj.filterCriteria = list;
+    setFilter(filterObj);
+  };
+  const onStartTimeChange = (value: string) => {
+    setTime(value, true, false);
+  };
+  const onStartTimeFormatChange = (value: string) => {
+    setTime(value, true, true);
+  };
+  const onEndTimeChange = (value: string) => {
+    setTime(value, false, false);
+  };
+  const onEndTimeFormatChange = (value: string) => {
+    setTime(value, false, true);
+  };
+  const onChangeStartDate = (value: string) => {
+    const filterObj = { ...filter };
+    filterObj.addedDate.startDate = value;
+    setFilter(filterObj);
+  };
+  const onChangeEndDate = (value: string) => {
+    const filterObj = { ...filter };
+    filterObj.addedDate.endDate = value;
+    setFilter(filterObj);
+  };
+  const {
+    lastSeen,
+    addedDate,
+    deviceId,
+    deviceType,
+    status,
+    filterCriteria
+  } = filter;
+  return (
+    <>
+      <Form>
+        <FormGroup label="Device ID" fieldId="filter-device-id">
+          <TextInput
+            isRequired
+            type="text"
+            id="device-id-filter-input"
+            name="device-id"
+            aria-describedby="Device Id for filter"
+            value={deviceId}
+            onChange={onChangeDeviceId}
+          />
+        </FormGroup>
+        <FormGroup label="Device Type" fieldId="filter-device-type">
+          <DropdownWithToggle
+            id={"device-type"}
+            name="Device Type"
+            className={css(styles.dropdown_align)}
+            toggleClass={css(styles.dropdown_toggle_align)}
+            position={DropdownPosition.left}
+            onSelectItem={value => onTypeSelect(value)}
```

Comment:
we don't need arrow function here. just  call function without arrow function , just make sure function params should match.
Logically nothing is wrong here but i will cause of performance, every time function instance will create. 
 onSelectItem={onTypeSelect}